rust/src/state_machine/phases/sum.rs

```diff
@@ -95,6 +103,18 @@ where
             coordinator_state,
             request_rx,
         } = self;
+
+        #[cfg(feature = "metrics")]
+        let Self {
+            inner: Sum {
+                sum_dict,
+                seed_dict,
+            },
+            coordinator_state,
+            request_rx,
+            metrics_tx,
+        } = self;
```

Comment:
any reason here it's not
```rust
let Self {
            inner: Sum {
                sum_dict,
                seed_dict,
            },
            coordinator_state,
            request_rx,
            #[cfg(feature = "metrics")] metrics_tx,
        } = self;
```
?

Reply:
I tried but to my surprise it doesn't work in the destructuring pattern.